src/cursor/abstract_cursor.ts

```diff
@@ -0,0 +1,697 @@
+            docs.push(...internalDocs);
+          }
+
+          fetchDocs();
+        });
+      };
+
+      fetchDocs();
+    });
+  }
+
+  // DO THESE PROPERTIES BELONG HERE?
+
+  /**
+   * Add a cursor flag to the cursor
+   *
+   * @param flag - The flag to set, must be one of following ['tailable', 'oplogReplay', 'noCursorTimeout', 'awaitData', 'partial' -.
+   * @param value - The flag boolean value.
+   */
+  addCursorFlag(flag: CursorFlag, value: boolean): this {
+    if (!CURSOR_FLAGS.includes(flag)) {
+      throw new MongoError(`flag ${flag} is not one of ${CURSOR_FLAGS}`);
+    }
+
+    if (typeof value !== 'boolean') {
+      throw new MongoError(`flag ${flag} must be a boolean value`);
+    }
+
+    this[kOptions][flag] = value;
+    return this;
+  }
+
+  /**
+   * Map all documents using the provided function
+   *
+   * @param transform - The mapping transformation method.
+   */
+  map(transform: (doc: Document) => Document): this {
+    const oldTransform = this[kTransform];
+    if (oldTransform) {
+      this[kTransform] = doc => {
+        return transform(oldTransform(doc));
+      };
+    } else {
+      this[kTransform] = transform;
+    }
+
+    return this;
+  }
+
+  /**
+   * Set the ReadPreference for the cursor.
+   *
+   * @param readPreference - The new read preference for the cursor.
+   */
+  setReadPreference(readPreference: ReadPreferenceLike): this {
+    if (readPreference instanceof ReadPreference) {
+      this[kOptions].readPreference = readPreference;
+    } else if (typeof readPreference === 'string') {
+      this[kOptions].readPreference = ReadPreference.fromString(readPreference);
+    } else {
+      throw new TypeError('Invalid read preference: ' + readPreference);
+    }
+
+    return this;
+  }
+
+  /**
+   * Set a maxTimeMS on the cursor query, allowing for hard timeout limits on queries (Only supported on MongoDB 2.6 or higher)
+   *
+   * @param value - Number of milliseconds to wait before aborting the query.
+   */
+  maxTimeMS(value: number): this {
+    if (typeof value !== 'number') {
+      throw new TypeError('maxTimeMS must be a number');
+    }
+
+    this[kOptions].maxTimeMS = value;
+    return this;
+  }
+
+  /**
+   * Set the batch size for the cursor.
+   *
+   * @param value - The number of documents to return per batch. See {@link https://docs.mongodb.com/manual/reference/command/find/|find command documentation}.
+   */
+  batchSize(value: number): this {
+    if (this[kOptions].tailable) {
+      throw new MongoError('Tailable cursors do not support batchSize');
+    }
+
+    if (typeof value !== 'number') {
+      throw new TypeError('batchSize requires an integer');
+    }
+
+    this[kOptions].batchSize = value;
+    return this;
+  }
+
+  /* @internal */
+  abstract _initialize(
+    session: ClientSession | undefined,
+    callback: Callback<ExecutionResult>
+  ): void;
+
+  /* @internal */
+  _getMore(batchSize: number, callback: Callback<Document>): void {
```

Comment:
We could also consider still making this protected and add a `//@ts-expect-error` comment to the nextDocument function.

Reply:
Hm, let's hold off on figuring this protected stuff out until a subsequent PR. Maybe we can also try storing `initialize` and `getMore` as properties passed in through options, and stored with symbols to avoid the method showing up in autocomplete